### PR TITLE
fixed evaluation of properties named property

### PR DESCRIPTION
**Pull request by machinekoder**
Fixes https://github.com/qmlweb/qmlweb/issues/346

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -277,8 +277,10 @@ function qmlweb_parse($TEXT, document_type, exigent_mode) {
       }
     } else if (S.token.type == "name") {
       if (S.token.value == "property" && (S.token.type == "name" || S.token.value == "var")) {
-        next();
-        return qmlpropdef();
+        if (!is_token(peek(), "punc", ":")) {
```
Comment (ChALkeR) [inline src/api.js]: Hm. Linter didn't catch this for some reason, but there is no need for nested ifs here, just use a single one:

``` js
      if (S.token.value == "property" &&
          (S.token.type == "name" || S.token.value == "var") &&
          !is_token(peek(), "punc", ":")) {
        next();
        return qmlpropdef();
      }
```

Upd: ah, we don't have linter enabled in this repo.

Comment (ChALkeR) [reply src/api.js]: Btw, `S.token.value == "property" && (S.token.type == "name" || S.token.value == "var")` could be reduced to `S.token.value == "property" && S.token.type == "name"`, perhaps this means that there is some other bug somewhere.

**Comment by ChALkeR** [comment]
LGTM with a nit (unnecessary `if` nesting).

Thanks!

File: src/api.js (hunk 1 above)
Comment (machinekoder) [reply src/api.js]: Good point, token.value should never be `property` and `var` at the same time.

**Comment by ChALkeR** [comment]
Thanks, landed in dfbcf70!

There was some more dead code, so I removed all those in a separate commit (eba88a9), and then rebased the actual change from this PR manually.

**Comment by machinekoder** [comment]
Awesome! When will you publish the next release?

**Comment by ChALkeR** [comment]
@machinekoder 
qmlweb-parser — just published 0.3.1.
qmlweb — not sure yet, I was hoping to finish some more structural changes before 0.2.0, but I might postpone those and release 0.2.0 pretty soon instead.

**Pull request merged**